sfepy/terms/terms_multilinear.py

```diff
@@ -1529,3 +1522,160 @@ def get_function(self, mat, parameter, mode=None, term_mode=None,
         return self.make_function(
             'IK,s(k:l)->K', mat, parameter, diff_var=diff_var,
         )
+
+
+class ENonSymElasticTerm(ETermBase):
+    r"""
+    Elasticity term with non-symmetric gradient. The indices of matrix
+    :math:`D_{ijkl}` are ordered as
+    :math:`[11, 12, 13, 21, 22, 23, 31, 32, 33]` in 3D and as
+    :math:`[11, 12, 21, 22]` in 2D.
+
+    :Definition:
+
+    .. math::
+        \int_{\Omega} \ull{D} \nabla \ul{v} : \nabla \ul{u}
+
+    :Arguments 1:
+        - material : :math:`\ull{D}`
+        - virtual  : :math:`\ul{v}`
+        - state    : :math:`\ul{u}`
+
+    :Arguments 2:
+        - material    : :math:`\ull{D}`
+        - parameter_1 : :math:`\ul{w}`
+        - parameter_2 : :math:`\ul{u}`
+    """
+    name = 'de_nonsym_elastic'
+    arg_types = (('material', 'virtual', 'state'),
+                 ('material', 'parameter_1', 'parameter_2'))
+    arg_shapes = {'material': 'D2, D2', 'virtual': ('D', 'state'),
+                  'state': 'D', 'parameter_1': 'D', 'parameter_2': 'D'}
+    modes = ('weak', 'eval')
+
+    def get_function(self, mat, virtual, state, mode=None, term_mode=None,
+                     diff_var=None, **kwargs):
+        fun = self.make_function(
+            'IK,n(i:j)->I,n(k:l)->K', mat, virtual, state, diff_var=diff_var,
```

Comment:
Is not this just `'ijkl,i.j,k.l'`, but `mat` needs to be reshaped to the fourth order? The reshaping could be done without a copy, right? Manually here...